api/types/database.go

```diff
@@ -484,6 +493,10 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 	if d.Spec.Protocol == "" {
 		return trace.BadParameter("database %q protocol is empty", d.GetName())
 	}
+	if d.IsDynamoDB() {
+		// DynamoDB gets its own checking logic for its unusual config.
+		return trace.Wrap(d.checkAndSetDynamoDBDefaults())
```

Comment:
Can we rename the `checkAndSetDynamoDBDefaults` to something like `handleDynamoDBConfig`  `process...`.  The `checkAndSet` default functions mostly are called at the beginning of function and  should not contain any complex "buisness logic".